lightning/src/util/ser_macros.rs

```diff
@@ -8,21 +8,54 @@
 // licenses.
 
 macro_rules! encode_tlv {
-	($stream: expr, {$(($type: expr, $field: expr)),*}) => { {
+	($stream: expr, {$(($type: expr, $field: expr)),*}, {$(($optional_type: expr, $optional_field: expr)),*}) => { {
```

Comment:
Not sure I fully understand the need for this. Aren't optional types defined by having an odd type number?